pkg/kubelet/kubelet.go

```diff
@@ -494,7 +501,8 @@ func NewMainKubelet(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps *Kub
 	klet.livenessManager = proberesults.NewManager()
 
 	klet.podCache = kubecontainer.NewCache()
-	klet.podManager = kubepod.NewBasicPodManager(kubepod.NewBasicMirrorClient(klet.kubeClient))
+	// podManager is also responsible for keeping secretManager contents up-to-date.
+	klet.podManager = kubepod.NewBasicPodManager(kubepod.NewBasicMirrorClient(klet.kubeClient), secretManager)
```

Comment:
If we expose the "register/unregister" functions to other components, eventually some other components may start calling these functions and the logic may grow even more complicated. If the these operations are only used by PodManager, PodManager can embed a new interface to expose `GetSecret` (or `GetConfigMap`) only. This is just a thought to see if we can hide unnecessary details to have tighter control.  I am okay with a separate secret manager as well.